content/en/blog/attribute-autocomplete-enhancement/index.md

```diff
@@ -0,0 +1,306 @@
+---
+
+# Enhancing Attribute Autocomplete Functionality -  Auto Correction and Type Suggestion
+
+In this blog, we discuss the two new enhancement added to the attribute name autocomplete feature in Texera, auto attribute correction and attribute type suggestion, making the feature more user-friendly.
+
+## Background
+
+### What are operator properties
+
+Texera workflows consist of sequences of operators. Every operator has properties as the parameters of the data process function. For example, in Fig. 1, the "Filter" operator has three properties, `Attribute`, `Condition`, and `Value`. The property values in Fig. 1 indicate that the operator filters the records with amount greater than 1.
+
+**![](https://lh6.googleusercontent.com/lxGfMeDfnw14f4NskeiSVBlz3_CFUQYrQJ8l97GnDo5L5LXoZ4w8mxJUa9AnnOWn7FLlxO-kwmValePube-Hw21BlwuhzFLco_pPhdrJu7SHObcQXJEtug4h-oBDIqnXdGXfwYlerpGvkZyaZ84bK4s)**
+
+Fig. 1, Filter operator properties
+
+### What is operator input schema
+
+Various formats of data can be processed by Texera, and the format of data is represented by its schema, which is a set of attributes. In order to let every operator know what attributes of the data are available, each operator deduces its output schema using its input schema and propagates it to the next operator. This mechanism is called "schema propagation." Schema propagation is an important part of the system, as it is the base of many useful features such as operator validation and attribute autocomplete.
+
+### What is attribute autocomplete
+
+Many operators have properties that take an attribute name of the input data as value. For example, the `Attribute` property in the "Sentiment Analysis" operator takes the name of the attribute on which it will perform analysis. Autocomplete feature allows users to select an input attribute from a dropdown menu instead of manually typing in the name.
+
+**![](https://lh3.googleusercontent.com/vqgcIM-ae7mRCIbah-ZyZRJkM4T4MMNdo3HGQtVT0RFfeh9J69z7SIUspMfY1IUjJ0V30U84PLdBMd3QouXKLEbfjUBc3Ac5q3_CtKk5XAP55I3QY0oG2U6Q11TIWcXUrVfjhbGpUUgV7LPp1R4ACeM)**
+
+Fig. 2, Dropdown menu for attribute autocomplete
+
+## Auto Attribute Correction
+
+### Motivation
+
+In the previous autocomplete implementation, when an attribute name in the input schema of an operator changes, the field of this operator that selects the attribute will not update its value according to the change in the attribute name, and the operator will therefore become invalid. Intuitively, it is expected that the field will be updated accordingly. 
+
+Therefore, we introduce the auto attribute correction feature that automatically corrects the attribute selection in operators whenever an attribute is deleted or renamed.
+
+### Demonstration
+
+The auto attribute correction feature updates the attribute selection in an operator under the following circumstances
+
+1. when the name of the attribute selected in the operator changes,
+2. when the attribute selected in the operator is deleted
```

Comment:
use the same terminology (i.e., `rename` and `deleted`) to refer to those two circumstatnces.

Also, one more question: if an attribute's type is changed, what happens? (I remember the "old design" can already accommodate this situation.  Please explicitly mention the behavior.

Reply:
I will explicitly explain it there. Since the auto attribute correction service only corrects the attribute property value (the autocomplete value), so it will do nothing if only the type is changed.